### PR TITLE
Add chain gas price API

### PR DESCRIPTION
**Pull request by matthew1001**
Closes https://github.com/hyperledger/firefly-transaction-manager/issues/55

Signed-off-by: Matthew Whitehead <matthew1001@gmail.com>

### CODE REVIEW

File: pkg/fftm/gas_management.go
Comment (peterbroadhurst) [inline pkg/fftm/gas_management.go:27]: Just wondering why we embed `ffcapi. GasPriceEstimateResponse ` in a wrapper `apitypes. LiveGasPrice` rather than just using `GasPriceEstimateResponse` directly?

**Review by peterbroadhurst — APPROVED** [review]
Looks great. Approved - one comment to consider before merging.

File: pkg/fftm/gas_management.go
Comment (matthew1001) [reply pkg/fftm/gas_management.go:27]: Yeah we seem to have a convention of wrapping `ffacpi` structures in an `apitype` structure when the API call involves a call down into the connector. So e.g. `getLiveStatus()` gets an `ffcapi.LiveResponse` structure back from the connector and wraps it in an `apitypes.LiveStatus` structure. I suspect it's a difference between APIs that call down into the `ffcapi` endpoint and those which generate their responses from FFTM itself. If it's OK I'll merge for now but if we think there's an unnecessary level of abstraction across some of the APIs I'd be happy to go through them all and remove the wrapper layer?

**Pull request merged**
